DependencyInjection/Compiler/SerializerConfigurationPass.php

```diff
@@ -33,13 +33,26 @@ public function process(ContainerBuilder $container)
         }
 
         if ($container->has('jms_serializer.serializer')) {
-            $container->setAlias('fos_rest.serializer', 'jms_serializer.serializer');
+            $container->setAlias('fos_rest.serializer', 'fos_rest.serializer.jms');
         } else {
             $container->removeDefinition('fos_rest.serializer.exception_wrapper_serialize_handler');
         }
 
         if ($container->has('serializer')) {
-            $container->setAlias('fos_rest.serializer', 'serializer');
+            $serializer = $container->findDefinition('serializer');
+            $class = $serializer->getClass();
+
+            if ('%' === $class[0]) {
+                $class = $container->getParameter(substr($class, 1, -1));
+            }
+
+            if (is_subclass_of($class, 'Symfony\Component\Serializer\SerializerInterface')) {
+                $container->setAlias('fos_rest.serializer', 'fos_rest.serializer.symfony');
+            } elseif (is_subclass_of($class, 'JMS\Serializer\SerializerInterface')) {
+                $container->setAlias('fos_rest.serializer', 'fos_rest.serializer.jms');
+            } else {
+                throw new \LogicException(sprintf('Cannot handle serializer of type "%s"', $class));
```

Comment:
well, to support everything, the compiler pass should also accept configuring a service implementing directly the FOSRestBundle interface